WalletWasabi/Blockchain/Keys/HdPubKey.cs

```diff
@@ -88,6 +97,10 @@ public double AnonymitySet
 	[JsonProperty(Order = 4)]
 	public KeyState KeyState { get; private set; }
 
+	[JsonProperty(Order = 5)]
+	[JsonConverter(typeof(ByteArrayJsonConverter))]
+	public byte[]? KeyIdBytes { get; }
```

Comment:
Is this nullable for backward compatibility?

Reply:
Yes. But I will check your solution and see if changing the equality function solves the problem.